### PR TITLE
feat(internal-plugin-encryption): add methods for Bring Your Own Master Key feature

### PR DESCRIPTION
**Pull request by yahcheng**
add methods which will be used by Control Hub for Bring Your Own Master Key feature. 
This is the related [wiki](https://sqbu-github.cisco.com/pages/WebExSquared/arch-docs/services/kms/proposals/kms-1028.html#controlhub-customer-kms-byok-protocol-examples) and [ux](https://www.figma.com/file/KLRZi3s00MaVOpncDOVqzl/BYOK?node-id=269%3A114222) and [jira](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-318613)

# COMPLETES #< INSERT LINK TO ISSUE >

## This pull request addresses

< DESCRIBE THE CONTEXT OF THE ISSUE >

## by making the following changes

< DESCRIBE YOUR CHANGES >

1. add method to download public key;
2. add method to upload master key;
3. add method to activate master key;
4. add method to get all master keys;
5. delete one master key;

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

add integration tests to cover the whole flow: get public key and then upload master key and then get all master keys and then activate the uploaded master and then delete the master key.

### I certified that

- [ ] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/node_modules/@webex/internal-plugin-encryption/test/integration/spec/kms.js [hunk 1]
```diff
@@ -314,6 +316,66 @@ describe('Encryption', function () {
...
+        .then((activeRes) => {
+          expect(activeRes.customerMasterKeys[0].usageState).to.have.string('');
+        })
+        .then(() => webex.internal.encryption.kms.deleteCustomerMasterKey({keyId: uploadedkeyId, assignedOrgId: spock.orgId})));
+    });
+
+    describe('#fetchPublicKey()', () => {
+      it('requests public key from the KMS', () => webex.internal.encryption.kms.fetchPublicKey({assignedOrgId: spock.orgId})
+        .then((publicKey) => {
+          console.warn(`in test, the public key is ${JSON.stringify(publicKey)}`);
```
Comment (arun3528) [inline packages/node_modules/@webex/internal-plugin-encryption/test/integration/spec/kms.js:369]: dont need console logs in test unless for debugging

Comment (yahcheng) [reply packages/node_modules/@webex/internal-plugin-encryption/test/integration/spec/kms.js:369]: Thanks, will remove this finally.

File: packages/node_modules/@webex/internal-plugin-encryption/src/kms.js [hunk 2]
```diff
@@ -250,6 +251,147 @@ const KMS = WebexPlugin.extend({
...
+  */
+  /**
+   * get public key from kms
+   * @param {Object} options
+   * @param {UUID} options.assignedOrgId the orgId
+   * @returns {Promise.<FetchPublicKeyResponse>} response of get public key api
+   */
+  fetchPublicKey({assignedOrgId}) {
+    this.logger.info('kms: fetch public key for byok');
+    console.warn(`---start to call fetchPublicKey in src ${assignedOrgId}`);
```
Comment (robstax) [inline packages/node_modules/@webex/internal-plugin-encryption/src/kms.js:268]: remove console statements. @arun3528 @InteractiveTimmy we probably want to add prettier/lint rules for this...

File: packages/node_modules/@webex/internal-plugin-encryption/src/kms.js [hunk 3]
```diff
@@ -250,6 +251,147 @@ const KMS = WebexPlugin.extend({
       });
   },
 
+  /**
+  * @typedef {Object} FetchPublicKeyResponse
```
Comment (robstax) [inline packages/node_modules/@webex/internal-plugin-encryption/src/kms.js:255]: in general i think these comments could be more descriptive. what are the possible `status` values? what is `requestId` used for? is it telemetry somewhere? etc.

File: packages/node_modules/@webex/internal-plugin-encryption/test/.eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (robstax) [inline packages/node_modules/@webex/internal-plugin-encryption/test/.eslintrc.json:1]: why was this added?

**Review by robstax — COMMENTED** [review]
seems mostly fine to me. some additional JS comments would be helpful and remove console statements

**Comment by arun3528** [comment]
Any update on this ?

**Comment by sreenara** [comment]
Closing this branch as there has been no activity for over 1.5 years.
